server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -823,8 +823,13 @@ static ClusterState innerPutTemplate(final ClusterState currentState, PutRequest
         }
 
         for (Alias alias : request.aliases) {
-            AliasMetadata aliasMetadata = AliasMetadata.builder(alias.name()).filter(alias.filter())
-                .indexRouting(alias.indexRouting()).searchRouting(alias.searchRouting()).build();
+            AliasMetadata aliasMetadata = AliasMetadata.builder(alias.name())
+                .filter(alias.filter())
+                .indexRouting(alias.indexRouting())
+                .searchRouting(alias.searchRouting())
+                .writeIndex(alias.writeIndex())
```

Comment:
I actually don't think we should persist this one. Putting a template in place with `is_write_index: true` will be confusing because only a single index can be marked as the write index. I think we should instead throw an exception if `alias.writeIndex()` is set to `true` here (fail early rather than fail late)

Reply:
Makes perfect sense. I added a validation against the alias having `is_write_index: true` and updated the tests